src/browser/NativeMessagingHost.cpp

```diff
@@ -77,6 +77,11 @@ void NativeMessagingHost::run()
         QString serverPath = getLocalServerPath();
         QFile::remove(serverPath);
 
+        // Ensure that STDIN is not being listened when proxy is used
+        if (m_notifier->isEnabled()) {
+            m_notifier->setEnabled(false);
```

Comment:
Shouldn't this be handled by the base class?

Reply:
This is just a precaution. It handles a situation where proxy has been disabled but is enabled again.